main.tf

```diff
@@ -4,10 +4,54 @@ data "aws_eks_cluster" "this" {
 
 data "aws_region" "current" {}
 
+locals {
+  argocd_enabled     = length(var.argocd) > 0 ? 1 : 0
+  grafana_enabled    = var.grafana_enabled ? 1 : 0
+  prometheus_enabled = var.prometheus_enabled ? 1 : 0
+  thanos_enabled     = var.thanos_enabled ? 1 : 0
+  storage            = var.thanos_storage == "s3" ? 0 : 1
+  namespace          = coalescelist(var.namespace == "" && local.argocd_enabled > 0 ? [{ "metadata" = [{ "name" = var.namespace_name }] }] : kubernetes_namespace.this, [{ "metadata" = [{ "name" = var.namespace }] }])[0].metadata[0].name
+}
+
+module "iam_assumable_role_admin" {
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "~> v3.6.0"
+  create_role                   = true
+  role_name                     = "${data.aws_eks_cluster.this.id}_${local.thanos_name}"
+  provider_url                  = replace(data.aws_eks_cluster.this.identity.0.oidc.0.issuer, "https://", "")
+  role_policy_arns              = [aws_iam_policy.thanos.arn]
+  oidc_fully_qualified_subjects = ["system:serviceaccount:${local.namespace}:${local.thanos_name}"]
+  tags                          = var.tags
+}
+
+resource "aws_iam_policy" "thanos" {
+  name_prefix = "${data.aws_eks_cluster.this.id}-thanos-"
+  description = "EKS thanos-s3 policy for cluster ${data.aws_eks_cluster.this.id}"
+  policy = jsonencode(
+    {
+      Version = "2012-10-17",
+      Statement = [
+        {
+          Effect = "Allow",
+          Action = [
+            "s3:ListBucket",
+            "s3:GetObject",
+            "s3:DeleteObject",
+            "s3:PutObject",
+            "s3:CreateBucket",
+            "s3:DeleteBucket"
+          ],
+          Resource = [
+            "arn:aws:s3:::*/*", #TODO: Change to ${aws_s3_bucket.thanos[0].id} or hardcoded name
```

Comment:
this policy has a too wide set of permissions for that service, can we point to a certain S3 bucket with a path?